### ISSUE TITLE
How to change selected row background color?

### ISSUE THREAD

**Issue by anjumpaul** [gregnb/mui-datatables#448]
## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

| Tech         | Version |
|--------------|---------|
| Material-UI  |         |
| MUI-datatables  |         |
| React        |         |
| browser      |         |
| etc          |         |

**Comment by gabrielliwerant** [comment]
Look into the custom styling section in the docs: https://github.com/gabrielliwerant/mui-datatables#customize-styling.

The JSS needed would look like this:

```
MuiTableRow: {
  root: {
    '&$selected': {
      backgroundColor: 'yellow'
    }
  }
},
```

**Comment by ruthzalo** [comment]
and when is the selected state changes?

**Comment by gabrielliwerant** [comment]
@ruthzalo Can you rephrase? I'm not sure what's being asked. You want to know what selection means in the context of the table? If so, it's when you select a row via checkbox.

**Comment by NYannick** [comment]
Hi, i know you aren't a Typescript guy but i need to selected a row without checkbox and eventually change his background color. So actually selectableRowsOnClick doesn't type in MUIDataTableOptions.

**Comment by gabrielliwerant** [comment]
@NYannick This is a bit offtopic, but if you want type support, you'll need to add it yourself to the definitely typed library. You can achieve the look you want by hiding the checkbox via style overrides.

**Comment by hibc** [comment]
@NYannick 

I was sort of able to change the color of a row using a onRowClick and getElementById(`MUIDataTableBodyRow-${rowIndex}`).

But this will remove other style effect for the row. (ex, hover over a row, etc)

`
// Table options
options= {
    onRowClick: (rowData, rowIndex) => {
        changeRowColor(rowIndex.rowIndex);
    }

changeRowColor(index){
    let row = document.getElementById(`MUIDataTableBodyRow-${index}`);
    row.setAttribute('style', 'background: yellow');
}
`

**Comment by kublyblu** [comment]
Has anything changed regarding this?
https://github.com/gregnb/mui-datatables/issues/448#issuecomment-469430049 does not work

**Comment by SagarHassanS** [comment]
> Look into the custom styling section in the docs: https://github.com/gabrielliwerant/mui-datatables#customize-styling.
> 
> The JSS needed would look like this:
> 
> ```
> MuiTableRow: {
>   root: {
>     '&$selected': {
>       backgroundColor: 'yellow'
>     }
>   }
> },
> ```

Is there any Update on this because this is not working

**Comment by SagarHassanS** [comment]
> @NYannick
> 
> This might be a temporary solution, but I was sort of able to change the color of a row using a onRowClick and getElementById(`MUIDataTableBodyRow-${rowIndex}`) and this might remove other style effect of the row. (ex, hover over a row, etc)
> 
> Ah unfortunately, this would not work if you are sorting or filtering rows. :(
> 
> ```
> // Table options
> options= {
>     onRowClick: (rowData, rowIndex) => {
>         changeRowColor(rowIndex.rowIndex);
>     }
> }
> 
> changeRowColor(index){
>     let row = document.getElementById(`MUIDataTableBodyRow-${index}`);
>     row.setAttribute('style', 'background: yellow');
> }
> ```

This works when we have a single page, If we have multiple pages the row.setAttribute fails as we have a null value is there any way to get the DOM of the row based on pagenumber

**Comment by CrisTovillaGpac** [comment]
Worked for me!
```
MuiTableRow: {
  root: {
      '&.Mui-selected:hover': {
        backgroundColor: 'rgba(0, 0, 0, 0.04)'
      }
  }
},
```

**Comment by RebaekSoparkKitchen** [comment]
> > Look into the custom styling section in the docs: https://github.com/gabrielliwerant/mui-datatables#customize-styling.
> > The JSS needed would look like this:
> > ```
> > MuiTableRow: {
> >   root: {
> >     '&$selected': {
> >       backgroundColor: 'yellow'
> >     }
> >   }
> > },
> > ```
> 
> Is there any Update on this because this is not working

This solution works for me, just add !important.
`MuiTableRow: {
            root: {
            backgroundColor: 'yellow',
            "&$selected": {
              backgroundColor: 'green !important'
            },
          },
          hover: {
            "&$root": { "&:hover": { backgroundColor: red} },
          }
`

**Comment by RebaekSoparkKitchen** [comment]
> > Look into the custom styling section in the docs: https://github.com/gabrielliwerant/mui-datatables#customize-styling.
> > The JSS needed would look like this:
> > ```
> > MuiTableRow: {
> >   root: {
> >     '&$selected': {
> >       backgroundColor: 'yellow'
> >     }
> >   }
> > },
> > ```
> 
> Is there any Update on this because this is not working

I think this solution works for me. Just add !important.
```
MuiTableRow: {
  root: {
    '&$selected': {
      backgroundColor: 'yellow !important' 
    }
  }
},
```

**Comment by LiaqatSaeed** [comment]
Hi try using this. It works for me.
`MuiTableRow: {
  root: {
        "& .Mui-selected": {
          backgroundColor: "rgba(76, 175, 80, 0.5) !important"
        }
  }
},`

**Comment by ezerobledo91** [comment]
This configuration works for several components and properties. warehouse.dark is a property of my own project. 

` const tableTheme = useMemo(
    () =>
      createTheme({
        palette: {
          background: {
            default: warehouse.dark === false ? 'rgb(254,255,244)' : '#121212e6',
          },
          text: {
            primary: warehouse.dark === false ? '#777777' : '#ffffffb6',
          },
        },
        components: {
          MuiTableRow: {
            styleOverrides: {
              root: {
                '&:hover': {
                  backgroundColor: warehouse.dark === false ? '#e5f6fd !important' : '#000000e6 !important',
                },
                '&.Mui-selected': {
                  backgroundColor: warehouse.dark === false ? '#e5f6fd !important' : '#46afaf6b !important',
                },
              },
            },
          },
          MuiCheckbox: {
            styleOverrides: {
              root: {
                color: warehouse.dark === false ? '' : 'inherit !important',
              },
            },
          },
          MuiIconButton: {
            styleOverrides: {
              root: {
                color: warehouse.dark === false ? '' : 'inherit !important',
              },
            },
          },
          MuiMenu: {
            styleOverrides: {
              list: {
                backgroundColor: warehouse.dark === false ? '' : 'black !important',
              },
            },
          },
          MuiInput: {
            styleOverrides: {
              root: {
                '&:after': {
                  borderBottom: warehouse.dark === true && 'solid 2px #46afaf6b',
                },
              },
            },
          },
        },
      }),
    [warehouse.dark]
  )`
